### PR TITLE
Fixing diktat-gradle-plugin

### LINKED ISSUE #627: Diktat-gradle-plugin 0.1.6 doesn't detect inputs on Linux
## Describe the bug
0 files are checked, while everything works fine on windows
It's time for integration tests: #576

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Introduced functional tests for diktat-gradle-plugin
* Calculate code coverage for functional tests for diktat-gradle-plugin
* Skip gradle tests when maven is run with `-DskipTests=true`
* Updated gradle to 6.7.1

This pull request closes #627

### CODE REVIEW

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt [hunk 1]
```diff
@@ -68,10 +68,10 @@ open class DiktatJavaExecTaskBase @Inject constructor(
                 }
                 .files
                 .forEach {
-                    add("\"${it.path}\"")
+                    add(it.path)
```
Comment (kentr0w) [inline diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:71]: Why is this necessary?

File: diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        Assertions.assertEquals(TaskOutcome.FAILED, diktatCheckBuildResult.outcome)
+        Assertions.assertTrue(
+            result.output.contains("[HEADER_MISSING_OR_WRONG_COPYRIGHT]")
+        )
+    }
+
+    @Test
+    fun `should execute diktatCheck with explicit inputs`() {
+        buildFile.appendText(
+            """${System.lineSeparator()}
```
Comment (kentr0w) [inline diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt:58]: Is this due to different OS?

**Review by kentr0w — APPROVED** [review]
lgtm, but a few questions to clarify

File: diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    private lateinit var buildFile: File
+
+    @BeforeEach
+    fun setUp() {
+        testProjectDir.create()
+        File("../examples/gradle-kotlin-dsl").copyRecursively(testProjectDir.root)
+        File(testProjectDir.root, "build.gradle.kts").delete()
+        buildFile = testProjectDir.newFile("build.gradle.kts").apply {
+            writeText(
+                """
```
Comment (aktsay6) [inline diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt:24]: Why diKTat is not complaining on this indent?

**Review by aktsay6 — APPROVED** [review]
lgtm

File: diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt (hunk 1 above)
Comment (petertrr) [reply diktat-gradle-plugin/src/main/kotlin/org/cqfn/diktat/plugin/gradle/DiktatJavaExecTaskBase.kt:71]: Seems that ktlint doesn't remove quotes from path patterns, they should be removed by a shell when invoking it from CLI, but in JavaExec they are not needed.

File: diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt (hunk 2 above)
Comment (petertrr) [reply diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt:58]: Yes, to make tests for both win and Linux

File: diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt (hunk 3 above)
Comment (petertrr) [reply diktat-gradle-plugin/src/functionalTest/kotlin/org/cqfn/diktat/plugin/gradle/DiktatGradlePluginFunctionalTest.kt:24]: `(` is a token that increases indent by one `INDENT_SIZE`, so it's valid.

**Pull request merged**